### PR TITLE
Fix babel config issues

### LINKED ISSUE #247: Suppressed Babel plugin parsing error results in no linting
### What steps are needed to reproduce the bug?

Run `stylelint "src/**/*.tsx"` 

### Relevant dependencies
```json
        "@emotion/styled": "11.6.0",
        "@stylelint/postcss-css-in-js": "0.37.2",
        "stylelint": "14.0.0",
        "stylelint-config-standard": "24.0.0",
        "stylelint-declaration-strict-value": "1.8.0",
        "stylelint-formatter-pretty": "2.1.1"
```

`babel.config.js` for `development`
```javascript
{
    development: {
      presets: [
        '@babel/preset-env',
        '@babel/preset-typescript',
        '@babel/preset-react'
      ],
      plugins: [
        [
          '@emotion',
          {
            sourceMap: true,
            cssPropOptimization: false
          }
        ],
        'babel-plugin-macros',
        '@babel/plugin-proposal-object-rest-spread',
        '@babel/plugin-proposal-class-properties',
        '@babel/plugin-proposal-private-methods'
      ]
    }
}
```

`.stylelintrc.json`
```json
{
    "defaultSeverity": "warning",
    "extends": "stylelint-config-standard",
    "overrides": [
        {
            "customSyntax": "@stylelint/postcss-css-in-js",
            "files": ["src/**/*.tsx"]
        },
        {
            "customSyntax": "postcss-scss",
            "files": ["sass/*/*.scss"]
        }
    ],
    "plugins": ["stylelint-declaration-strict-value"],
    "rules": {
        "property-no-unknown": true,
        "scale-unlimited/declaration-strict-value": [
            ["/color/", "/font/", "background"],
            {
                "severity": "warning"
            }
        ]
    }
}
```

### What did you expect to happen?

I expect there to be warnings in the output of Stylelint for my css-in-js styles using Emotion.

### What actually happened?

There are no warnings found by Stylelint, and the linting finishes successfully. There are errors during parsing that are suppressed in the following code block:

https://github.com/stylelint/postcss-css-in-js/blob/09af2eb5a1ba41279e099d095c268e170f02e88b/extract.js#L160-L165

If I uncomment the `console.error` line, I see the following error in the babel parser for each source
```
TypeError: p is not iterable
    at <projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:10658:43
    at Array.some (<anonymous>)
    at hasPlugin (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:10654:18)
    at validatePlugins (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:10694:9)
    at getParser (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:15287:5)
    at parse (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:15232:22)
    at parser (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/core/lib/parser/index.js:52:34)
    at parser.next (<anonymous>)
    at Function.parse (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/core/lib/parse.js:31:37)
    at parse.next (<anonymous>)
```

This happened when I was upgrading from Stylelint 13 -> 14. When using v13, I was using the bundled css-in-js parser, but in v14 that was removed, so now I'm trying to use `@stylelint/postcss-css-in-js` as a custom syntax for my `.tsx` files that use Emotion for css-in-js. But I don't see the style warnings that I was seeing before because of this parsing error with Babel.

It looks like the issue is that one or more plugins are not valid in this function in the Babel parser
```javascript
function hasPlugin(plugins, expectedConfig) {
  const [expectedName, expectedOptions] = typeof expectedConfig === "string" ? [expectedConfig, {}] : expectedConfig;
  const expectedKeys = Object.keys(expectedOptions);
  const expectedOptionsIsEmpty = expectedKeys.length === 0;
  return plugins.some(p => {
    if (typeof p === "string") {
      return expectedOptionsIsEmpty && p === expectedName;
    } else {
      const [pluginName, pluginOptions] = p;

      if (pluginName !== expectedName) {
        return false;
      }

      for (const key of expectedKeys) {
        if (pluginOptions[key] !== expectedOptions[key]) {
          return false;
        }
      }

      return true;
    }
  });
}
```

but I'm not sure why that would be.

### PR DESCRIPTION
**Pull request by hudochenkov**
> Which issue, if any, is this issue related to?

Fixes #247 and #250

> Is there anything in the PR that needs further explanation?

Issues was in incorrect config object passed to Babel, if there is existing user Babel config with plugins/presets. As @astone123 [pointed out](https://github.com/stylelint/postcss-css-in-js/issues/247#issuecomment-999650865).

I've updated Babel to the latest version just in case. It caused adding `index` in JSONs.

### CODE REVIEW

**Comment by hudochenkov** [comment]
@ybiquitous @jeddy3 folks, maybe you have time to check this PR? It would really helped me out, as stylelint doesn't work in my project ¯﻿\﻿_﻿(﻿ツ﻿)﻿_﻿/﻿¯

**Comment by ybiquitous** [comment]
@hudochenkov OK. Honestly, I'm not so familiar with this project, but anyway I try to review it. 👀

File: extract.js [hunk 1]
```diff
@@ -144,7 +144,14 @@ function loadBabelOpts(opts) {
 
 		if (Array.isArray(fileOpts[key]) && Array.isArray(opts.parserOpts[key])) {
 			// combine arrays for plugins
-			opts.parserOpts[key] = opts.parserOpts[key].concat(fileOpts[key]);
+			// plugins in fileOpts could be string, array or object
+			for (const plugin of fileOpts[key]) {
+				if (Array.isArray(plugin) || typeof plugin === 'string') {
+					opts.parserOpts[key].push(plugin);
+				} else {
+					opts.parserOpts[key].push([plugin.key, plugin.options]);
```
Comment (ybiquitous) [inline extract.js:152]: [ask] The original code uses `.concat` not to mutate the `opts.parserOpts[key]` array, while the changed code uses `.push` to mutate the array. No problems?

Comment (hudochenkov) [reply extract.js]: Done.

**Comment by hudochenkov** [comment]
> Honestly, I'm not so familiar with this project

@ybiquitous Me neither :)

Thank you for your review!

**Review by ybiquitous — APPROVED** [review]
@hudochenkov Looks good, thank you 👍🏼

**Comment by ybiquitous** [comment]
@hudochenkov Do you want to release the next version?

**Comment by hudochenkov** [comment]
@ybiquitous yes, I'll release it later today. Thank you!

**Comment by hudochenkov** [comment]
> - Fixed: silent fail if Babel config is present ([#258](https://github.com/stylelint/postcss-css-in-js/pull/258)).

**Pull request merged**